services/QuillLMS/client/app/bundles/Staff/components/comprehension/activitySessions/sessionsIndex.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import { Link } from 'react-router-dom';
+import { useQuery } from 'react-query';
+import * as moment from 'moment'
+
+import { DataTable, Error, Spinner, DropdownInput } from '../../../../Shared/index';
+import { fetchActivity, fetchActivitySessions } from '../../../utils/comprehension/activityAPIs';
+import { DropdownObjectInterface } from '../../../interfaces/comprehensionInterfaces';
+
+const quillCheckmark = 'https://assets.quill.org/images/icons/check-circle-small.svg';
+
+const SessionsIndex = ({ match }) => {
+  const { params } = match;
+  const { activityId } = params;
+
+  const [pageNumber, setPageNumber] = React.useState<DropdownObjectInterface>(null);
+  const [dropdownOptions, setDropdownOptions] = React.useState<DropdownObjectInterface[]>(null);
+  const pageNumberForQuery = pageNumber && pageNumber.value ? pageNumber.value : 1;
+
+  // cache activity data for updates
+  const { data: activityData } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
+    queryFn: fetchActivity
+  });
+
+  // cache activity sessions data for updates
+  const { data: sessionsData } = useQuery({
+    queryKey: [`activity-${activityId}-sessions`, activityId, pageNumberForQuery],
+    queryFn: fetchActivitySessions
+  });
+
+  React.useEffect(() => {
+    sessionsData && !dropdownOptions && getDropdownOptions(sessionsData)
+  }, [sessionsData]);
+
+  function handlePageChange(number) {
+    setPageNumber(number)
+  }
+
+  function formatSessionsData(activitySessions: any[]) {
+    return activitySessions.map(session => {
+      const formattedSession = {...session};
+      const { start_date, session_uid, because_attempts, but_attempts, so_attempts, complete } = session;
+      const dateObject = new Date(start_date);
+      const date = moment(dateObject).format("MM/DD/YY");
+      const time = moment(dateObject).format("HH:MM A");
+      const total = because_attempts + but_attempts + so_attempts;
+      formattedSession.id = session_uid;
+      formattedSession.date = date;
+      formattedSession.time = time;
+      formattedSession.because_attempts = because_attempts;
+      formattedSession.but_attempts = but_attempts;
+      formattedSession.so_attempts = so_attempts;
+      formattedSession.total_attempts = total;
+      formattedSession.view_link = <Link className="data-link" to={`/activities/${activityId}/activity-sessions/${session_uid}/overview`}>View</Link>;
+      formattedSession.completed = complete ? <img alt="quill-circle-checkmark" src={quillCheckmark} /> : "";
+      return formattedSession;
```

Comment:
This one is less confusing because you're only defining the one object, but I would probably still format this as a literal.

Reply:
yeah not sure why I did it like this, I think it's because I was checking at each step-- updating